### PR TITLE
working on all endpoints and input validation

### PR DESCRIPTION
**Pull request by Nadinefiona**
all endpoints and test

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+const authRoute = require('./routes/auth');
+const postRoute = require('./routes/posts');
+const res = require('express/lib/response');
+
+dotenv.config();
+
+
+
+//Connect to DB
+mongoose.connect('mongodb+srv://nadine_1:fiona123@cluster0.6zhup.mongodb.net/database?retryWrites=true&w=majority',() =>{
```
Comment (habaumugisha1) [inline index.js:41]: use variable environments to hide the sensitive data

File: index.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  console.log('connected to db!')
+});
+
+//Middlewares
+app.use(express.json());
+
+
+//Route Middlewares
+app.use('/api/user', authRoute);
```
Comment (habaumugisha1) [inline index.js:50]: where are those authRoute and post

**Comment by Nadinefiona** [comment]
Check There are  in other file in folder of model and routes

**Comment by Nadinefiona** [comment]
Check There are  in other file in folder of model and routes

**Review by habaumugisha1 — COMMENTED** [review]
can you use this chain of router -> validation middleware -> controller -> model
you are mixing business logic into routes

**Pull request merged**